pkg/rbac/node_rbac.go

```diff
@@ -0,0 +1,272 @@
+package rbac
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	core "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	k8sapierrors "k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// EnsureNodeSpecificRBAC creates node-specific RBAC resources with proper resourceNames restriction
+func EnsureNodeSpecificRBAC(ctx context.Context, ctrlRuntimeClient client.Client, k8sClient kubernetes.Interface, wmcoNamespace, nodeName string) error {
+	if nodeName == "" {
+		return fmt.Errorf("node name cannot be empty")
+	}
+
+	// Create node-specific ServiceAccount
+	if err := ensureNodeSpecificServiceAccount(ctx, k8sClient, wmcoNamespace, nodeName); err != nil {
+		return err
+	}
+
+	// Create node-specific ClusterRole with resourceNames restriction
+	if err := ensureNodeSpecificClusterRole(ctx, k8sClient, nodeName); err != nil {
+		return err
+	}
+
+	// Create node-specific ClusterRoleBinding
+	if err := ensureNodeSpecificClusterRoleBinding(ctx, k8sClient, wmcoNamespace, nodeName); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// CleanupNodeSpecificRBAC removes node-specific RBAC resources for a deleted node
+func CleanupNodeSpecificRBAC(ctx context.Context, ctrlRuntimeClient client.Client, k8sClient kubernetes.Interface, wmcoNamespace, nodeName string) error {
+	// Clean up ServiceAccounts by label (handles both node-name and IP-based ServiceAccounts)
```

Comment:
If users can override node names, how  to check they do not allow label injection or resource name spoofing?